src/test/pmem2_config/pmem2_config.c

```diff
@@ -0,0 +1,199 @@
+	ut_pmem2_delete_cfg(&cfg);
+}
+
+/*
+ * test_set_rw_fd - test setting O_RDWR fd
+ */
+static void
+test_set_rw_fd(const char *file)
+{
+	struct pmem2_config *cfg;
+
+	int fd = OPEN(file, O_RDWR);
+
+	ut_pmem2_alloc_cfg(&cfg);
+	ut_pmem2_set_fd(cfg, fd);
+	ut_pmem2_delete_cfg(&cfg);
+
+	CLOSE(fd);
+}
+
+/*
+ * test_set_ro_fd - test setting O_RDONLY fd
+ */
+static void
+test_set_ro_fd(const char *file)
+{
+	struct pmem2_config *cfg;
+
+	int fd = OPEN(file, O_RDONLY);
+
+	ut_pmem2_alloc_cfg(&cfg);
+	ut_pmem2_set_fd(cfg, fd);
+	ut_pmem2_delete_cfg(&cfg);
+
+	CLOSE(fd);
+}
+
+/*
+ * test_set_negative - test setting negative fd
+ */
+static void
+test_set_negative_fd(const char *unused)
+{
+	struct pmem2_config *cfg;
+
+	ut_pmem2_alloc_cfg(&cfg);
+	ut_pmem2_set_fd(cfg, -42);
+	ut_pmem2_delete_cfg(&cfg);
+}
+
+/*
+ * test_set_invalid_fd - test setting invalid fd
+ */
+static void
+test_set_invalid_fd(const char *file)
+{
+	struct pmem2_config *cfg;
+
+	/* open and close the file to get invalid fd */
+	int fd = OPEN(file, O_WRONLY);
+	CLOSE(fd);
+
+	ut_pmem2_alloc_cfg(&cfg);
+	if (pmem2_config_set_fd(cfg, fd) != PMEM2_E_INVAL)
+		UT_FATAL("pmem2_config_set_fd: %s", pmem2_errormsg());
+
+	ut_pmem2_delete_cfg(&cfg);
+}
+
+/*
+ * test_set_wronly_fd - test setting wronly fd
+ */
+static void
+test_set_wronly_fd(const char *file)
+{
+	struct pmem2_config *cfg;
+
+	int fd = OPEN(file, O_WRONLY);
+	ut_pmem2_alloc_cfg(&cfg);
+	if (pmem2_config_set_fd(cfg, fd) != PMEM2_E_INVALID_HANDLE)
+		UT_FATAL("pmem2_config_set_fd: %s", pmem2_errormsg());
+
+	ut_pmem2_delete_cfg(&cfg);
+	CLOSE(fd);
+}
+
+/*
+ * test_cfg_alloc_enomem - test pmem2_config allocation with error injection
+ */
+static void
+test_cfg_alloc_enomem(const char *unused)
+{
+	struct pmem2_config *cfg;
+	if (!common_fault_injection_enabled()) {
+		return;
+	}
+	common_inject_fault_at(PMEM_MALLOC, 1, "Zalloc");
+	if (pmem2_config_new(&cfg) != PMEM2_E_NOMEM)
+		UT_FATAL("pmem2_config_new: %s", pmem2_errormsg());
+
+	UT_ASSERTeq(cfg, NULL);
+}
+
+/*
+ * test_delete_null_config - test pmem2_delete on NULL config
+ */
+static void
+test_delete_null_config(const char *unused)
+{
+	struct pmem2_config *cfg = NULL;
+	/* should not crash */
+	ut_pmem2_delete_cfg(&cfg);
+}
+
+typedef void (*test_fun)(const char *file);
+
+static struct test_list {
+	const char *name;
+	test_fun test;
+} list[] = {
+	{"alloc_cfg", test_cfg_alloc},
+	{"set_rw_fd", test_set_rw_fd},
+	{"set_ro_fd", test_set_ro_fd},
+	{"set_negative_fd", test_set_negative_fd},
+	{"set_invalid_fd", test_set_invalid_fd},
+	{"set_wronly_fd", test_set_wronly_fd},
+	{"alloc_cfg_enomem", test_cfg_alloc_enomem},
+	{"delete_null_config", test_delete_null_config},
+};
+
+int
+main(int argc, char **argv)
+{
+	START(argc, argv, "pmem2_config");
+	if (argc != 3)
+		UT_FATAL("usage: %s file test_case", argv[0]);
+
+	char *file = argv[1];
+	char *test_case = argv[2];
+
+	for (int i = 0; i < ARRAY_SIZE(list); i++) {
+		if (strcmp(list[i].name, test_case) == 0) {
```

Comment:
do you *need* to execute these tests in separate processes? Do they operate on some global state?